dali/pipeline/operators/color/hsv.h

```diff
@@ -0,0 +1,281 @@
+    hsv::assign_argument_value<Backend>(spec, hsv::kHue, hue_);
+    hsv::assign_argument_value<Backend>(spec, hsv::kSaturation, saturation_);
+    hsv::assign_argument_value<Backend>(spec, hsv::kValue, value_);
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, batch_size_);
+    }
+  }
+
+  ~Hsv() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(Hsv);
+
+ protected:
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+  USE_OPERATOR_MEMBERS();
+  hsv::argument_t<Backend> hue_, saturation_, value_;
+  DALIDataType output_type_;
+  kernels::KernelManager kernel_manager_;
+};
+
+
+
+class HsvCpu : public Hsv<CPUBackend> {
+ public:
+  explicit HsvCpu(const OpSpec &spec) : Hsv(spec) {}
+
+  /*
+   * So that compiler wouldn't complain, that
+   * "overloaded virtual function `dali::Operator<dali::CPUBackend>::RunImpl`
+   * is only partially overridden in class `dali::HsvCpu`"
+   */
+  using Operator<CPUBackend>::RunImpl;
+
+  ~HsvCpu() override = default;
+
+
+  DISABLE_COPY_MOVE_ASSIGN(HsvCpu);
+
+
+ protected:
+  bool SetupImpl(std::vector<::dali::OutputDesc> &output_desc,
+                 const workspace_t<CPUBackend> &ws) override;
+
+
+  void RunImpl(Workspace<CPUBackend> &ws) override;
+
+
+ private:
+  mat3 transformation_matrix(float hue, float saturation, float value) const {
+    using namespace hsv;  // NOLINT
+    return Rgb2Yiq * compose_hue(hue) * compose_saturation(saturation) * compose_value(value) *
+           Yiq2Rgb;
+  }
+
+
+  template <typename Kernel, typename InputType>
+  kernels::TensorListShape<> CallSetup(const TensorVector<CPUBackend> &input, int instance_idx) {
+    kernels::KernelContext ctx;
+    kernels::TensorListShape<> sh = input.shape();
+    std::vector<kernels::TensorShape<>> shapes;
+    shapes.resize(sh.num_samples());
+    for (size_t i = 0; i < shapes.size(); i++) {
+      const auto tvin = view<const InputType, 3>(input[i]);
+      const auto tmat = transformation_matrix(hue_, saturation_, value_);
+      const auto reqs = kernel_manager_.Setup<Kernel>(instance_idx, ctx, tvin, tmat);
+      kernels::TensorListShape<> out_sh = reqs.output_shapes[0];
+      shapes[i] = out_sh.tensor_shape(0);
+    }
+    return {shapes};
+  }
+};
+
+
+
+class HsvGpu : public Hsv<GPUBackend> {
+ public:
+  explicit HsvGpu(const OpSpec &spec) : Hsv(spec) {}
+
+
+  ~HsvGpu() override = default;
+
+
+  DISABLE_COPY_MOVE_ASSIGN(HsvGpu);
+
+
+ protected:
+  bool SetupImpl(std::vector<::dali::OutputDesc> &output_desc,
+                 const workspace_t<GPUBackend> &ws) override;
+
+
+  void RunImpl(Workspace<GPUBackend> &ws) override;
+
+
+ private:
+  /**
+   * @brief Creates transformation matrices based on given args
+   */
+  std::vector<mat3>
+  determine_transformation(const std::vector<float> &hue, const std::vector<float> &saturation,
```

Comment:
1. A matrix is a "transform", a "transformation" is the act of applying the transform.
```suggestion
  get_transform_matrices(const std::vector<float> &hue, const std::vector<float> &saturation,
```

Reply:
https://english.stackexchange.com/questions/5454/transform-or-transformation
I'd say both would be correct